crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -79,9 +110,36 @@ impl EthereumSettlementClient {
         EthereumSettlementClient { provider, core_contract_client, wallet, wallet_address }
     }
 
+    #[cfg(test)]
+    pub fn with_test_settings(settings: &impl SettingsProvider, provider: RootProvider<Http<Client>>) -> Self {
+        let settlement_cfg: EthereumSettlementConfig = settings.get_settings(SETTLEMENT_SETTINGS_NAME).unwrap();
+
+        let private_key = get_env_var_or_panic(ENV_PRIVATE_KEY);
+        let signer: PrivateKeySigner = private_key.parse().expect("Failed to parse private key");
+        let wallet_address = signer.address();
+        let wallet = EthereumWallet::from(signer);
+
+        let fill_provider = Arc::new(
+            ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(settlement_cfg.rpc_url),
+        );
+
+        let core_contract_address = if *SHOULD_IMPERSONATE_ACCOUNT {
```

Comment:
TL:DR : will destroy `single source of truth practice of code`.

`SHOULD_IMPERSONATE_ACCOUNT` is used in `with_test_settings` and `update_state_with_blob`.

If we try this way, we'll be sending a new param to `with_test_settings` which is alright since it's testing code, 
and we don't change anything on `update_state_with_blob` since it's production code, what we end up with is two functions requiring same data but taking it from two difference source, resulting in multiple source of same truth, leading more management for the same.